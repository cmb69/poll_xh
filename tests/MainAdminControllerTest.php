<?php

namespace Poll;

use ApprovalTests\Approvals;
use PHPUnit\Framework\TestCase;
use Plib\View;

class MainAdminControllerTest extends TestCase
{
    public function testRendersPollResults(): void
    {
        $dataService = $this->createStub(DataService::class);
        $dataService->method("getPollNames")->willreturn(["foo"]);
        $dataService->method("findPoll")->willReturn($this->poll());
        $sut = new MainAdminController(
            $dataService,
            new View("./views/", XH_includeVar("./languages/en.php", "plugin_tx")["poll"])
        );
        Approvals::verifyHtml($sut->defaultAction());
    }

    private function poll(): Poll
    {
        $poll = new Poll();
        $poll->setEndDate(1147483647);
        $poll->setVoteCount("Germany", 1);
        $poll->setVoteCount("Brazil", 0);
        $poll->setTotalVotes(1);
        return $poll;
    }
}