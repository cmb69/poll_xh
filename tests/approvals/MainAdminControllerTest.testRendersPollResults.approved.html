<div id="poll_admin">
<h1>foo</h1>

<p>The poll has ended.</p>
<h5>The Results</h5>
<div class="poll_results">
    <div class="poll_result">
        <div class="poll_text">Germany – 100% (1 vote)</div>
        <div class="poll_bar" style="width: 100%">&nbsp;</div>
    </div>
    <div class="poll_result">
        <div class="poll_text">Brazil – 0% (0 votes)</div>
        <div class="poll_bar" style="width: 0%">&nbsp;</div>
    </div>
</div>
<p><strong>In total 1 person has voted.</strong></p>
</div>
